bottom_sheet/lib/main.dart

```diff
@@ -0,0 +1,154 @@
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  // This widget is the root of your application.
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Bottom Sheet',
+      theme: ThemeData(
+        primarySwatch: Colors.red,
+        visualDensity: VisualDensity.adaptivePlatformDensity,
+      ),
+      home: MyHomePage(title: 'Bottom Sheet'),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  MyHomePage({Key key, this.title}) : super(key: key);
+
+  final String title;
+
+  @override
+  _MyHomePageState createState() => _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(widget.title),
+        backgroundColor: Colors.redAccent,
+      ),
+      body: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            new RaisedButton(
+              child: Text("Bottom Sheet"),
+              onPressed: () {
+                _bottomSheetMore(context);
+              },
+            )
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+void _bottomSheetMore(context) {
+  showModalBottomSheet(
+    context: context,
+    builder: (builder) {
+      return new Container(
+        padding: EdgeInsets.only(
+          left: 5.0,
+          right: 5.0,
+          top: 5.0,
+          bottom: 5.0,
+        ),
+        decoration: new BoxDecoration(
+            color: Colors.white,
+            borderRadius: new BorderRadius.only(
+                topLeft: const Radius.circular(10.0),
+                topRight: const Radius.circular(10.0))),
+        child: new Wrap(
+          children: <Widget>[
+            new ListTile(
+              title: const Text(
+                'More',
+                style: TextStyle(
+                  fontSize: 14.0,
+                  fontWeight: FontWeight.w700,
+                ),
+              ),
+            ),
+            new ListTile(
+              leading: new Container(
+                width: 4.0,
+                child: Icon(
+                  Icons.favorite,
+                  color: Colors.pink,
+                  size: 24.0,
+                ),
+              ),
+              title: const Text(
+                'Favourites',
+                style: TextStyle(
+                  fontSize: 14.0,
+                  fontWeight: FontWeight.w700,
+                ),
+              ),
+            ),
```

Comment:
Each `ListTile`'s code is same. Create a function to abstract that and pass in arguments to the function so you can construct a `ListTile`.

```
ListTile getListTile(icon, iconColor, titleText) {
  return new ListTile(
    leading: new Container(
      width: 4.0,
      child: Icon(
        icon,
        color: iconColor,
        size: 24.0,
      ),
    ),
    title: new Text(
      titleText,
      style: TextStyle(
        fontSize: 14.0,
        fontWeight: FontWeight.w700,
      ),
    ),
  );
}
```

and then use as:

```
 children: <Widget>[
            getListTile(Icons.favorite, Colors.pink, "More"),
            getListTile(Icons.favorite, Colors.pink, "Favourites")
            ]
```